cmd/libs/go2idl/conversion-gen/generators/conversion.go

```diff
@@ -693,13 +705,22 @@ func (g *genConversion) doStruct(inType, outType *types.Type, sw *generator.Snip
 				sw.Do("out.$.name$ = $.outType|raw$(in.$.name$)\n", args)
 			}
 		case types.Map, types.Slice, types.Pointer:
+			if g.isDirectlyAssignable(m.Type, outMember.Type) {
+				sw.Do("out.$.name$ = in.$.name$\n", args)
```

Comment:
Will need to look into go2idl - it's possible we don't expose this code path today.